snuba/admin/migrations_policies.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+from functools import wraps
+from typing import Any, Callable, Dict
+
+from flask import Response, jsonify, make_response
+from werkzeug.routing import BaseConverter
+
+from snuba import settings
+from snuba.migrations.policies import (
+    AllMigrationsPolicy,
+    MigrationPolicy,
+    NoMigrationsPolicy,
+    NonBlockingMigrationsPolicy,
+)
+from snuba.migrations.runner import MigrationKey
+
+policy_map = {
+    "AllMigrationsPolicy": AllMigrationsPolicy(),
+    "NoMigrationsPolicy": NoMigrationsPolicy(),
+    "NonBlockingMigrationsPolicy": NonBlockingMigrationsPolicy(),
+}
+
+
+class MigrationActionConverter(BaseConverter):
+    regex = r"(?:run|reverse)"
+
+
+def check_migration_perms(f: Callable[..., Response]) -> Callable[..., Response]:
+    @wraps(f)
+    def check_group_perms(*args: Any, **kwargs: Any) -> Response:
+        ADMIN_ALLOWED_MIGRATION_GROUPS: Dict[str, MigrationPolicy] = {
+            k: policy_map[v] for k, v in settings.ADMIN_ALLOWED_MIGRATION_GROUPS.items()
```

Comment:
should we handle the `KeyError` here if someone sets a policy in settings that's not in the `policy_map`?

Reply:
the keys are set in `settings.ADMIN_ALLOWED_MIGRATION_GROUPS` in the settings file. `policy_map` is a static field that maps the policy name to class. It's not to be changed by the user. moved it inside to the function to make that clear